### PR TITLE
feat: Rebrand to Revenge

### PR DESCRIPTION
**Pull request by Gabe-W-J**
Altered logo and strings that contained "bunny" to match revenge. String changes need double checking to make sure I didn't mess up something important. If I did I will re-do the branch as per request.

P.S: When Revenge cannot load, the alert refers to it as Bunny, I went to go change this alert, but found no mention of it in the code?

### CODE REVIEW

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
Looks good. A readme, issue templates and so on should be added.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
-    "name": "revenge",
-    "private": true,
+    "name": "@revenge-mod/revenge",
+    "exports": {
+        "types": "./dist/types/index.d.ts"
```
Comment (PalmDevs) [inline package.json:4]: Need to generate typings here for this to work, can be done in `dev`.

File: src/lib/api/native/loader.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { Theme } from "@lib/themes";
+import { Theme } from "@lib/addons/themes";
```
Comment (PalmDevs) [inline src/lib/api/native/loader.ts:1]: Merged was done incorrectly in `dev`, so I corrected it here.

File: docs/2_building.md
Comment (PalmDevs) [inline docs/2_building.md:1]: Need to improve this, can also be done in `dev`.

**Review by PalmDevs — APPROVED** [review]
LGTM

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+name: Release
+on:
+    push:
+        branches: [main, dev]
+
+jobs:
+    build:
+        name: Build and release
```
Comment (oSumAtrIX) [inline .github/workflows/release.yml:8]: Just "Release" is simpler. It implies building already.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+            uses: actions/checkout@v4
+
+          - uses: actions/setup-node@v4
+            with:
+                node-version: 22
+
+          - name: Install pnpm and dependencies
+            run: |
+                npm i -g pnpm
+                pnpm i
```
Comment (oSumAtrIX) [inline .github/workflows/release.yml:24]: There should be an action for pnpm.

File: .releaserc [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    "@semantic-release/release-notes-generator",
+    "@semantic-release/changelog",
+    [
+      "@semantic-release/git",
+      {
+        "assets": [
+          "README.md",
+          "CHANGELOG.md"
+        ],
+        "message": "chore(release): ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}"
```
Comment (oSumAtrIX) [inline .releaserc:26]:
```suggestion
        "message": "chore: Release ${nextRelease.version} [skip ci]\n\n${nextRelease.notes}" 
```

File: README.md [hunk 6]
```diff
@@ -1,42 +1,72 @@
...
+
+## 💪 Features
+
+- **🔌 Plugins**: Extend Discord with custom features
+- **🎨 Themes & Fonts**: Customize Discord's appearance to your liking
+- **🧪 Experiments**: Try out Discord's new features before they're rolled out
+
+## ⬇️ Download
 
 ### Android
```
Comment (oSumAtrIX) [inline README.md:47]: Perhaps a paragraph would be good here which says that this repository is for internal use and if they want to use Revenge on Android they should use Revenge Manager or Revenge Xposed.

File: README.md [hunk 7]
```diff
@@ -1,42 +1,72 @@
...
 
-- **Root** with Xposed - [RevengeXposed](https://github.com/revenge-mod/RevengeXposed/releases/latest)
-- **Non-root** - [Revenge Manager](https://github.com/revenge-mod/RevengeManager/releases/latest)
+- **🩹 Root with Xposed**: [RevengeXposed](https://github.com/revenge-mod/revenge-xposed/releases/latest)
+- **📵 Non-root**: [Revenge Manager](https://github.com/revenge-mod/revenge-manager/releases/latest)
 
 ### iOS
-- [**RevengeTweak**](https://github.com/revenge-mod/RevengeTweak) - Get prebuilt rootful and rootless `.deb` files or the prepatched `.ipa `
+- [**RevengeTweak**](https://github.com/revenge-mod/revenge-tweak): Prebuilt rootful and rootless `.deb` files or the prepatched `.ipa `
```
Comment (oSumAtrIX) [inline README.md:52]: Format MD
```suggestion

- [**RevengeTweak**](https://github.com/revenge-mod/revenge-tweak): Prebuilt rootful and rootless `.deb` files or the prepatched `.ipa `
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
+# 👋 Contribution guidelines
+
+This document describes how to contribute to Revenge.
```
Comment (oSumAtrIX) [inline CONTRIBUTING.md:3]: Should this refer to Revenge bundle or Revenge? The other repos refer to Revenge Manager or Xposed respectively.

File: README.md [hunk 9]
```diff
@@ -1,42 +1,72 @@
...
+Thank you for considering contributing to Revenge.
+You can find the contribution guidelines [here](CONTRIBUTING.md).
 
-    ```
-    pnpm i
-    ```
+### 🛠️ Building
 
-5. Build Revenge's code
+To build a Revenge, you can follow the [documentation](/docs).
```
Comment (oSumAtrIX) [inline README.md:64]:
```suggestion
To build Revenge, you can follow the [documentation](/docs).
```

File: docs/2_building.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
+## Building
+1. Install a Revenge loader with config support (any mentioned in the [Installing](#installing) section).
+2. Go to **Settings** > **General** and enable **Developer Settings**.
+3. Clone the repository
+
+    ```sh
+    git clone https://github.com/revenge-mod/Revenge.git
```
Comment (oSumAtrIX) [inline docs/2_building.md:7]: Need to update the URL to revenge-bundle.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
lgtm apart from the review.

File: .github/workflows/release.yml (hunk 4 above)
Comment (PalmDevs) [reply .github/workflows/release.yml:24]: What do you mean by that?

File: CONTRIBUTING.md (hunk 8 above)
Comment (PalmDevs) [reply CONTRIBUTING.md:3]: I'm not really sure. What do you think?

File: README.md (hunk 9 above)
Comment (PalmDevs) [reply README.md:72]: I think it's best to rewrite the whole file, not in this PR though.

File: docs/2_building.md (hunk 10 above)
Comment (PalmDevs) [reply docs/2_building.md:7]: I think it's best to rewrite the whole file, not in this PR though.

File: .github/workflows/release.yml (hunk 4 above)
Comment (oSumAtrIX) [reply .github/workflows/release.yml:24]: https://github.com/pnpm/action-setup

File: CONTRIBUTING.md (hunk 8 above)
Comment (oSumAtrIX) [reply CONTRIBUTING.md:3]: I'd say bundle so that it is consistent with the other reops and true to its name

File: README.md (hunk 9 above)
Comment (oSumAtrIX) [reply README.md:72]: The issue can be addressed here for the time being

File: docs/2_building.md (hunk 10 above)
Comment (oSumAtrIX) [reply docs/2_building.md:7]: The issue can be addressed now and the file rewrited in another PR

**Review by PalmDevs — APPROVED** [review]
I think this LGTM.

File: .github/workflows/release.yml [hunk 11]
```diff
@@ -14,14 +14,19 @@ jobs:
...
-          - name: Install pnpm and dependencies
-            run: |
-                npm i -g pnpm
-                pnpm i
+          - name: Install pnpm
+            uses: pnpm/action-setup@v4
+            with:
+                version: latest
+          
+          - name: Install dependencies
```
Comment (oSumAtrIX) [inline .github/workflows/release.yml:28]: Is this step necessary? The action i would assume would've done that already

File: README.md [hunk 12]
```diff
@@ -1,42 +1,79 @@
...
+
+## 💪 Features
+
+- **🔌 Plugins**: Extend Discord with custom features
+- **🎨 Themes & Fonts**: Customize Discord's appearance to your liking
+- **🧪 Experiments**: Try out Discord's new features before they're rolled out
+
+## ⬇️ Download
+
+This repository releases JavaScript bundles for loaders to execute. These are the official Revenge loaders:
```
Comment (oSumAtrIX) [inline README.md:51]: Could simplify this and say that to use Revencd you should use the loaders below

File: README.md [hunk 13]
```diff
@@ -1,42 +1,79 @@
...
+- **📵 Non-root**: [Revenge Manager](https://github.com/revenge-mod/revenge-manager/releases/latest)
 
 ### iOS
-- [**RevengeTweak**](https://github.com/revenge-mod/RevengeTweak) - Get prebuilt rootful and rootless `.deb` files or the prepatched `.ipa `
 
-## Building
-1. Install a Revenge loader with config support (any mentioned in the [Installing](#installing) section).
-2. Go to **Settings** > **General** and enable **Developer Settings**.
-3. Clone the repository
+- [**RevengeTweak**](https://github.com/revenge-mod/revenge-tweak): Prebuilt rootful and rootless `.deb` files or the prepatched `.ipa`
```
Comment (oSumAtrIX) [inline README.md:60]: Might wanna keep an eye on the pre-patched ones

**Review by oSumAtrIX — APPROVED** [review]

**Pull request merged**